src/main/java/org/openmaptiles/layers/Housenumber.java

```diff
@@ -59,13 +65,63 @@ public class Housenumber implements
   Tables.OsmHousenumberPoint.Handler,
   ForwardingProfile.FeaturePostProcessor {
 
-  public Housenumber(Translations translations, PlanetilerConfig config, Stats stats) {}
+  private static final Logger LOGGER = LoggerFactory.getLogger(Housenumber.class);
+  private static final Character OSM_SEPARATOR = ';';
+  private static final String DISPLAY_SEPARATOR = "–";
+  private static final Pattern NO_CONVERSION_PATTERN = Pattern.compile("[^0-9;]");
+  private final Stats stats;
+
+  public Housenumber(Translations translations, PlanetilerConfig config, Stats stats) {
+    this.stats = stats;
+  }
+
+  private static String displayHousenumberNonumeric(String[] numbers) {
+    return numbers[0]
+      .concat(DISPLAY_SEPARATOR)
+      .concat(numbers[numbers.length - 1]);
+  }
+
+  protected static String displayHousenumber(String housenumber) {
+    if (housenumber.indexOf(OSM_SEPARATOR) < 0) {
+      return housenumber;
+    }
+
+    String[] numbers = Arrays.stream(housenumber.split(OSM_SEPARATOR.toString()))
+      .filter(n -> !n.trim().isEmpty())
+      .toArray(String[]::new);
+    if (numbers.length <= 0) {
+      // not much to do with strange/invalid entries like "3;" or ";" etc.
+      return housenumber;
+    }
+
+    Matcher matcher = NO_CONVERSION_PATTERN.matcher(housenumber);
+    if (matcher.find()) {
+      return displayHousenumberNonumeric(numbers);
+    }
+
+    // numeric display house number
+    var statistics = Arrays.stream(numbers)
+      .collect(Collectors.summarizingInt(Integer::parseUnsignedInt));
+    return String.valueOf(statistics.getMin())
+      .concat(DISPLAY_SEPARATOR)
+      .concat(String.valueOf(statistics.getMax()));
+  }
 
   @Override
   public void process(Tables.OsmHousenumberPoint element, FeatureCollector features) {
+    String housenumber;
+    try {
+      housenumber = displayHousenumber(element.housenumber());
```

Comment:
I see that openmaptiles [uses street and block_number tags now](https://github.com/openmaptiles/openmaptiles/compare/v3.14..master#diff-281b2691711d0047d6db7b21fe9284dbf748a7669f91411465be94f76c43cbfdR23) although I can't quite decipher why? Do we need to do anything with them here?

Reply:
Porting done here: https://github.com/phanecak-maptiler/planetiler-openmaptiles/pull/23

Already merged also into this.